test/unittests/UserTest.php

```diff
@@ -0,0 +1,526 @@
+<?php
+/**
+ * Unit tests for User class
+ *
+ * PHP Version 5
+ *
+ * @category Tests
+ * @package  Main
+ * @author   Alexandra Livadas <alexandra.livadas@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+require_once __DIR__ . '/../../vendor/autoload.php';
+use PHPUnit\Framework\TestCase;
+/**
+ * Unit test for User class
+ *
+ * @category Tests
+ * @package  Main
+ * @author   Alexandra Livadas <alexandra.livadas@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class UserTest extends TestCase
+{
+    /**
+     * The username for the private user object
+     *
+     * @var string
+     */
+    private $_username;
+    /**
+     * Stores user information
+     *
+     * @var array
+     */
+    private $_userInfo
+        = array('ID'                     => 1,
+                'UserID'                 => '968775',
+                'Password'               => 'pass123',
+                'Real_name'              => 'John Doe',
+                'First_name'             => 'John',
+                'Last_name'              => 'Doe',
+                'Degree'                 => 'Eng.D',
+                'Position_title'         => 'Doctor',
+                'Institution'            => 'MCIN',
+                'Department'             => 'Neuroscience',
+                'Address'                => '123 Main St',
+                'City'                   => 'Montreal',
+                'State'                  => 'Quebec',
+                'Zip_code'               => '123',
+                'Country'                => 'Canada',
+                'Phone'                  => '123-456-7890',
+                'Fax'                    => null,
+                'Email'                  => 'john.doe@mcgill.ca',
+                'Privilege'              => 1,
+                'PSCPI'                  => 'Y',
+                'DBAccess'               => '123',
+                'Active'                 => 'Y',
+                'Password_hash'          => null,
+                'Password_expiry'        => '2020-07-16',
+                'Pending_approval'       => 'Y',
+                'Doc_Repo_Notifications' => 'Y',
+                'language_preference'    => 2,
+                'active_from'            => '2017-07-16',
+                'active_to'              => '2020-07-16'
+          );
+    /**
+     * The userInfo table that should result from calling the factory function
+     * TODO There should be a way to write this that takes up less space!
```

Comment:
You could set `$_userInfoComplete` equal to `$_userInfo` and then just push the new elements to the end:

```php
$_userInfoComplete['Sites']                  => 'psc_test;psc_test2';
...
```